src/ripple/rpc/handlers/LedgerData.cpp

```diff
@@ -137,36 +137,45 @@ doLedgerDataGrpc(
     grpc::Status status = grpc::Status::OK;
 
     std::shared_ptr<ReadView const> ledger;
-    if (RPC::ledgerFromRequest(ledger, context))
+    if (auto status = RPC::ledgerFromRequest(ledger, context))
     {
-        grpc::Status errorStatus{
-            grpc::StatusCode::NOT_FOUND, "ledger not found"};
+        grpc::Status errorStatus;
+        if (status.toErrorCode() == rpcINVALID_PARAMS)
+        {
+            errorStatus = grpc::Status(
+                grpc::StatusCode::INVALID_ARGUMENT, status.message());
+        }
+        else
+        {
+            errorStatus =
+                grpc::Status(grpc::StatusCode::NOT_FOUND, status.message());
+        }
         return {response, errorStatus};
     }
 
     ReadView::key_type key = ReadView::key_type();
```

Comment:
Question for my own understanding: Why are you using `ReadView::key_type` here, but using `uint256::fromVoid` later in the function?

They [are the same thing](https://github.com/ripple/rippled/blob/develop/src/ripple/ledger/ReadView.h/#L198). Why not just declare `key` and `stopKey` as `ripple::uint256`'s? Can you call `ReadView::key_type::fromVoid`?

Reply:
No good reason. Probably something that started as a copy paste from somewhere else. I'll switch this to `ripple::uint256`, since I think that is clearer.